src/mono/wasm/build/WasmApp.Common.targets

```diff
@@ -174,6 +175,7 @@
     <!-- VS uses DeployOnBuild, and sdk sets _IsPublishing -->
     <WasmBuildOnlyAfterPublish Condition="'$(WasmBuildOnlyAfterPublish)' == '' and ('$(DeployOnBuild)' == 'true' or '$(_IsPublishing)' == 'true')">true</WasmBuildOnlyAfterPublish>
     <WasmGenerateAppBundle Condition="'$(WasmGenerateAppBundle)' == '' and '$(OutputType)' != 'Library'">true</WasmGenerateAppBundle>
+    <WasmGenerateAppBundle Condition="'$(WasmGenerateAppBundle)' == '' and '$(IsLibraryMode)' == 'true'">true</WasmGenerateAppBundle>
```

Comment:
nit: It can be combined with the previous condition (`empty and (output not library or isLibraryMode)`)